kite-data/kite-data-crunch/src/main/java/org/kitesdk/data/crunch/DatasetTarget.java

```diff
@@ -85,7 +85,14 @@ public boolean handleExisting(WriteMode writeMode, long lastModForSource,
       }
     }
 
-    boolean exists = !view.isEmpty();
+    boolean ready = false;
+    try {
+      ready = view.isReady();
+    } catch (UnsupportedOperationException ex) {
+      // view isn't readiable, continue as if it wasn't ready
+    }
+    // a view exists if it isn't empty, or if it has been marked ready
+    boolean exists = !view.isEmpty() || ready;
```

Comment:
Sounds like `signalReady` is basically the same thing as a FS "touch": it causes the view to exist if it did not already and updates the modified time. Sounds good to me. We will need to make sure the behavior is correct for both MR and Crunch.